libs/libpcache/src/faults.cpp

```diff
@@ -0,0 +1,215 @@
+#include <faults/faults.hpp>
+
+using namespace std;
+
+namespace faults {
+
+Fault::Fault(string type) {
+    this->type = type;
+}
+
+Fault::~Fault(){}
+
+
+// Torn seq fault
+ReorderF::ReorderF(string op, vector<int> persist, int occurrence) : Fault(TORN_SEQ) {
+    (this->counter).store(0);
+    this->op = op;
+    this->persist = persist;
+    this->occurrence = occurrence;
+    (this->group_counter).store(0);
+
+}
+
+ReorderF::ReorderF() : Fault(TORN_SEQ) {
+	vector <int> v;
+	(this->counter).store(0);
+    (this->group_counter).store(0);
+    this->op = "";
+	this->persist = v;
+    this->occurrence = 0;
+}
+
+ReorderF::~ReorderF(){}
+
+bool ReorderF::check(string op, vector<int> persist) {
+    if (op != "write") return false;
+
+    bool check = true;
+    for (auto & p : persist) {
+        if (p <= 0) {
+            check = false;
+            break;
+        }
+    }
+    return check;
+}
+
+vector<string> ReorderF::check_with_errors() {
```

Comment:
```suggestion
vector<string> ReorderF::validate() {
```

Reply:
I'll update this section based on your suggestion